ocaml/xapi-idl/cluster/cluster_interface.ml

```diff
@@ -15,12 +15,51 @@ type debug_info = string [@@deriving rpcty]
 (** Name of the cluster *)
 type cluster_name = string [@@deriving rpcty]
 
-(** An IPv4 address (a.b.c.d) *)
-type address = IPv4 of string [@@deriving rpcty]
+type ip = IPv4 of string | IPv6 of string [@@deriving rpcty]
```

Comment:
This is quite the golden opportunity to use [Ipaddr.t](https://ocaml.org/p/ipaddr/latest/doc/Ipaddr/index.html#type-t) as it gives quite a bit of functionality that is now coded ad hoc as we use string for representation. Did you consider using this type? If so, was it too complex to adapt it here?

Reply:
looks like Ipaddr.t is not rpcty'able? We would need to serialse and deserialse it each time we use it, seems a bit too complex?